test/main/modules.js

```diff
@@ -49,6 +49,30 @@ QUnit.test( "does not run before subsequent tests", function( assert ) {
 	assert.equal( this.beforeCount, 1, "beforeCount did not increase from last test" );
 } );
 
+QUnit.module( "before (skip)", {
+	before: function( assert ) {
+		assert.ok( true, "before hook ran" );
+
+		if ( typeof this.beforeCount === "undefined" ) {
+			this.beforeCount = 0;
+		}
+
+		this.beforeCount++;
+	}
+} );
+
+QUnit.skip( "first test in module is skipped" );
+
+QUnit.test( "runs before first unskipped test", function( assert ) {
+	assert.expect( 2 );
+	assert.equal( this.beforeCount, 1, "beforeCount should be one" );
```

Comment:
We can ditch the `beforeCount` stuff. Since we `assert.expect()` we know whether or not the hook ran.